cmd/index/bundles/command.go

```diff
@@ -172,12 +174,18 @@ func run(cmd *cobra.Command, args []string) error {
 		log.Errorf("unable to inspect the index image: %s", err)
 	}
 
-	if err := actions.ExtractIndexDB(flags.IndexImage, flags.ContainerEngine); err != nil {
+	if err := actions.ExtractIndexDBorCatalogs(flags.IndexImage, flags.ContainerEngine); err != nil {
 		return err
 	}
 
 	log.Info("Gathering data...")
-	reportData, err = getDataFromIndexDB(reportData)
+
+	// check here to see if it's index.db or file-based catalogs
+	if isFBC() {
+		reportData, err = getDataFromFBC(reportData)
+	} else {
+		reportData, err = getDataFromIndexDB(reportData)
+	}
```

Comment:
You may want to check out `action.Render` which supports rendering FBC structs from both SQLite and FBC images.

https://pkg.go.dev/github.com/operator-framework/operator-registry@v1.26.2/alpha/action#Render